sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -867,89 +897,85 @@ export class MessageReceiver extends LinkEntity {
         return;
       }
 
-      // We should attempt to reopen only when the receiver(sdk) did not initiate the close
-      let shouldReopen = false;
-      if (receiverError && !wasCloseInitiated) {
-        const translatedError = translate(receiverError) as MessagingError;
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is retryable. This is a candidate for re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        shouldReopen = true;
+      const translatedError = receiverError ? translate(receiverError) : receiverError;
+
+      // Track-1
+      //   - We should only attempt to reopen if either no error was present,
+      //     or the error is considered retryable.
+      // Track-2
+      //  Reopen
+      //   - If no error was present
+      //   - If the error is considered retryable
+      //   - Any errors that were triggered by a disconnect,
+      //     we don't care if the retryable flag is set to true on the error
+      //     because `@azure/core-amqp` is stricter in setting the retryable flag to true.
+      //   - More details here - https://github.com/Azure/azure-sdk-for-js/pull/8580#discussion_r417087030
+      const shouldReopen =
+        !translatedError ||
+        (isMessagingError(translatedError) ? translatedError.retryable : false) ||
+        causedByDisconnect;
```

Comment:
To truly mimic Track 1, I believe we need to respect the retryable flag if the error was a messaging error...
In Track 1, the errors that were neither amqp error nor system errors fell into the last bucket of "treat everything as retryable".

```suggestion
      // Track-1
      //   - We should only attempt to reopen if either no error was present,
      //     or the error is considered retryable.
      // Track-2
      //  Reopen
      //   - If no error was present
      //   - If the error is a MessagingError and is considered retryable
      //   - Any non MessagingError that was triggered by a disconnect because such errors do not get
      //     translated by `@azure/core-amqp` to a MessagingError and we need to do our best when
      //     it comes to connection being disconnected
      //   - More details here - https://github.com/Azure/azure-sdk-for-js/pull/8580#discussion_r417087030
      let shouldReopen = true;
      if (isMessagingError(translatedError)) {
          shouldReopen = translatedError.retryable;
      } else {
          shouldReopen = !translatedError || causedByDisconnect
      }
```